autogenerated.ts

```diff
@@ -9,19 +9,19 @@ namespace mlrunner {
     //% fixedInstance
     export const Circle = new MlEvent(4, "Circle");
 
-    actions = [None,Shake,Still,Circle];
+    actions = [None, Shake, Still, Circle];
 
     control.onEvent(MlRunnerIds.MlRunnerInference, 1, () => {
-      prevAction = 1;
+      maybeUpdateActionStats(None);
     });
     control.onEvent(MlRunnerIds.MlRunnerInference, 2, () => {
-      prevAction = 2;
+      maybeUpdateActionStats(Shake);
     });
     control.onEvent(MlRunnerIds.MlRunnerInference, 3, () => {
-      prevAction = 3;
+      maybeUpdateActionStats(Still);
     });
     control.onEvent(MlRunnerIds.MlRunnerInference, 4, () => {
-      prevAction = 4;
+      maybeUpdateActionStats(Circle);
     });
```

Comment:
Is this needed if it is also executed on `MlEvent.onEvent()`

Reply:
So, `MlEvent.onEvent()` code is only executed for all actions if the user has included the block for every action. If they are missing `on Circle start`, then the onEvent code will never run for the Circle action, even if it is inferred by the model on the device. 

There's a bit of pain getting this to work where the block is present and where the block is missing as well as working for the sim. I'll have another look at this because it can probably be nicer.

Reply:
Ah, right, that makes sense, thanks Rob!